### PR TITLE
Move viewer's time panel to its own crate & show number of events on component hover in Streams

### PR DESCRIPTION
**Pull request by Wumpf**
Moves the time panel to a separate crate.

Slight ui changes on hover to untangle it from the selection panel:

image

image

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)

PR Build Summary: https://build.rerun.io/pr/2233

### CODE REVIEW

**Review by emilk — APPROVED** [review]
Sweet

File: crates/re_time_panel/src/lib.rs [hunk 1]
```diff
@@ -537,24 +514,21 @@ impl TimePanel {
                 let is_visible = ui.is_rect_visible(full_width_rect);
 
                 if is_visible {
+                    let empty_messages_over_time = TimeHistogram::default();
+                    let messages_over_time = data
+                        .times
+                        .get(ctx.rec_cfg.time_ctrl.timeline())
+                        .unwrap_or(&empty_messages_over_time);
+
+                    let total_num_messages = messages_over_time.total_count();
```
[the remark this answers is not in the document]
Comment (Wumpf) [reply crates/re_time_panel/src/lib.rs:523]: checking and fixing if needed ⌛!

**Comment by nikolausWest** [comment]
Screenshot 2023-05-29 at 10 36 52

Remember to not mix using "events" and "messages". We use the term "events", which is more consistent with the terminology used in stream processing.

**Comment by Wumpf** [comment]
good catch. Doing a followup where I check for usage of "message"

**Comment by Wumpf** [comment]
This actually opens some interesting questions on the Rust api where we work with a `MsgSender`. Something to ponder for later rust api refactors

**Pull request merged**